bot/utils/regex.py

```diff
@@ -12,4 +12,11 @@
     r"(?P<invite>[a-zA-Z0-9\-]+)",                # the invite code itself
     flags=re.IGNORECASE
 )
+
 MESSAGE_ID_RE = re.compile(r'(?P<message_id>[0-9]{15,20})$')
+
+DISCORD_MESSAGE_LINK_RE = re.compile(
+    r"(https?:\/\/(?:(ptb|canary|www)\.)?discord(?:app)?\.com\/channels\/"
```

Comment:
There's a missing ?: before the `ptb`:

(https?:\/\/(?:( __?:__ ptb|canary|www)\.)?dis

That turns this into a two-group capture regex. As we don't have a use (afaict) here for that second half, we should add the ?: to simplify the interface. This will change the return result of .findall() to a simple list of entries.